From b6807bf457aa3debcd60c49908ebd1b147744e20 Mon Sep 17 00:00:00 2001
Message-ID: <b6807bf457aa3debcd60c49908ebd1b147744e20.1708289807.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Fri, 16 Feb 2024 15:17:56 +0000
Subject: [PATCH v2 01/10] gnu: go-github-com-alecthomas-chroma: Move to
 golang-xyz.

* gnu/packages/golang.scm (go-github-com-alecthomas-chroma): Move from
here ...
* gnu/packages/golang-xyz.scm: ... to here.

* gnu/packages/configuration-management.scm: Add (gnu packages
golang-xyz) module.

Change-Id: I0f5f7068f530b282f3636678e9edc4fad6c8c94a
---
 gnu/packages/configuration-management.scm |  1 +
 gnu/packages/golang-xyz.scm               | 33 ++++++++++++++++++++++-
 gnu/packages/golang.scm                   | 29 --------------------
 3 files changed, 33 insertions(+), 30 deletions(-)

diff --git a/gnu/packages/configuration-management.scm b/gnu/packages/configuration-management.scm
index 6632f429e2..d0105ea75c 100644
--- a/gnu/packages/configuration-management.scm
+++ b/gnu/packages/configuration-management.scm
@@ -23,6 +23,7 @@ (define-module (gnu packages configuration-management)
   #:use-module (gnu packages golang)
   #:use-module (gnu packages golang-build)
   #:use-module (gnu packages golang-web)
+  #:use-module (gnu packages golang-xyz)
   #:use-module (gnu packages version-control)
   #:use-module (gnu packages textutils)
   #:use-module ((guix licenses) #:prefix license:)
diff --git a/gnu/packages/golang-xyz.scm b/gnu/packages/golang-xyz.scm
index b0a5955405..024b38bd6e 100644
--- a/gnu/packages/golang-xyz.scm
+++ b/gnu/packages/golang-xyz.scm
@@ -3,15 +3,16 @@
 ;;; Copyright © 2018 Pierre-Antoine Rouby <pierre-antoine.rouby@inria.fr>
 ;;; Copyright © 2019 Brian Leung <bkleung89@gmail.com>
 ;;; Copyright © 2020 Efraim Flashner <efraim@flashner.co.il>
+;;; Copyright © 2020 Oleg Pykhalov <go.wigust@gmail.com>
 ;;; Copyright © 2021 Raghav Gururajan <rg@raghavgururajan.name>
 ;;; Copyright © 2021 Sarah Morgensen <iskarian@mgsn.dev>
 ;;; Copyright © 2022 Dominic Martinez <dom@dominicm.dev>
 ;;; Copyright © 2023 Benjamin <benjamin@uvy.fr>
 ;;; Copyright © 2023 Hilton Chain <hako@ultrarare.space>
 ;;; Copyright © 2023 Katherine Cox-Buday <cox.katherine.e@gmail.com>
-;;; Copyright © 2023, 2024 Sharlatan Hellseher <sharlatanus@gmail.com>
 ;;; Copyright © 2023 Thomas Ieong <th.ieong@free.fr>
 ;;; Copyright © 2023 Timo Wilken <guix@twilken.net>
+;;; Copyright © 2023, 2024 Sharlatan Hellseher <sharlatanus@gmail.com>
 ;;; Copyright © 2024 Artyom V. Poptsov <poptsov.artyom@gmail.com>
 ;;;
 ;;; This file is part of GNU Guix.
@@ -75,6 +76,36 @@ (define-public go-github-com-a8m-envsubst
 substitution.")
     (license license:expat)))
 
+(define-public go-github-com-alecthomas-chroma
+  (package
+    (name "go-github-com-alecthomas-chroma")
+    (version "0.8.0")
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/alecthomas/chroma")
+             (commit (string-append "v" version))))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32 "066a6rdmf670d3v5sc7chbn7db09ldgxjympb03pcqwk644dixb1"))))
+    (build-system go-build-system)
+    (arguments
+     `(#:import-path "github.com/alecthomas/chroma"))
+    (native-inputs
+     (list go-github-com-dlclark-regexp2
+           go-github-com-alecthomas-assert
+           go-github-com-alecthomas-colour
+           go-github-com-alecthomas-repr
+           go-github-com-mattn-go-isatty
+           go-github-com-sergi-go-diff))
+    (home-page "https://github.com/alecthomas/chroma/")
+    (synopsis "General purpose syntax highlighter in pure Go")
+    (description
+     "Chroma takes source code and other structured text and converts it into
+syntax highlighted HTML, ANSI-coloured text, etc.")
+    (license license:expat)))
+
 (define-public go-github-com-alecthomas-participle-v2
   (package
     (name "go-github-com-alecthomas-participle-v2")
diff --git a/gnu/packages/golang.scm b/gnu/packages/golang.scm
index bde0c74859..59d0df0781 100644
--- a/gnu/packages/golang.scm
+++ b/gnu/packages/golang.scm
@@ -7643,35 +7643,6 @@ (define-public go-github-com-sergi-go-diff
 @end itemize\n")
     (license license:expat)))
 
-(define-public go-github-com-alecthomas-chroma
-  (package
-    (name "go-github-com-alecthomas-chroma")
-    (version "0.8.0")
-    (source (origin
-              (method git-fetch)
-              (uri (git-reference
-                    (url "https://github.com/alecthomas/chroma")
-                    (commit (string-append "v" version))))
-              (file-name (git-file-name name version))
-              (sha256
-               (base32
-                "066a6rdmf670d3v5sc7chbn7db09ldgxjympb03pcqwk644dixb1"))))
-    (build-system go-build-system)
-    (arguments
-     `(#:import-path "github.com/alecthomas/chroma"))
-    (native-inputs
-     (list go-github-com-dlclark-regexp2
-           go-github-com-alecthomas-assert
-           go-github-com-alecthomas-colour
-           go-github-com-alecthomas-repr
-           go-github-com-mattn-go-isatty
-           go-github-com-sergi-go-diff))
-    (home-page "https://github.com/alecthomas/chroma/")
-    (synopsis "General purpose syntax highlighter in pure Go")
-    (description "Chroma takes source code and other structured text and
-converts it into syntax highlighted HTML, ANSI-coloured text, etc.")
-    (license license:expat)))
-
 (define-public go-github-com-muesli-reflow-wordwrap
   (package
     (name "go-github-com-muesli-reflow-wordwrap")

base-commit: fdbf4192f5eaa7fdb5e6e2e98ada0726c8104824
-- 
2.41.0

